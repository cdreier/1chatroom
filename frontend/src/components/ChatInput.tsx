import React, { useState } from 'react'
import styled from 'styled-components'

const InputContainer = styled.div`
  background-color: palegreen;
  padding: 12px;
  width: 100%;
  >form{
    display: flex;
  }
`

const Input = styled.input`
  flex: 1;
  padding: 6px;
  outline: none;
`

interface ChatInputProps {
  onSubmit: (msg: string) => void
}

const ChatInput: React.FC<ChatInputProps> = ({ onSubmit }) => {

  const [msg, setMsg] = useState('')

  const submit = (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault()
    e.stopPropagation()
    onSubmit(msg)
    setMsg('')
  }

  return (
    <InputContainer>
      <form onSubmit={(e) => submit(e)}>
        <Input value={msg} onChange={e => setMsg(e.target.value)} />
      </form>
    </InputContainer>
  )
}

export default ChatInput
