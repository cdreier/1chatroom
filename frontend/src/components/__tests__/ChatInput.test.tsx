import React from 'react'
import renderer from 'react-test-renderer'
import { render, fireEvent } from '@testing-library/react'
import ChatInput from '../ChatInput'
import { ThemeProvider } from 'styled-components'
import Theme from '../../theme/default'

describe('ChatInput', () => {

  test('it renders', () => {
    const sub = jest.fn()
    const component = renderer.create(
      <ThemeProvider theme={Theme}>
        <ChatInput onSubmit={sub}  />
      </ThemeProvider>,
    ).toJSON()
    expect(component).toMatchSnapshot()
  })

  test('that submit event triggers', () => {
    const sub = jest.fn()
    const { getByTestId } = render(
      <ThemeProvider theme={Theme}>
        <ChatInput onSubmit={sub}  />
      </ThemeProvider>,
    )
    fireEvent.input(getByTestId('input'), { target: { value: 'yay' } })
    fireEvent.submit(getByTestId('input'))
    expect(sub).toBeCalledWith('yay')
  })

  test('that submit event does not trigger if msg is empty', () => {
    const sub = jest.fn()
    const { getByTestId } = render(
      <ThemeProvider theme={Theme}>
        <ChatInput onSubmit={sub}  />
      </ThemeProvider>,
    )
    fireEvent.submit(getByTestId('input'))
    expect(sub).not.toHaveBeenCalled()
  })

})
