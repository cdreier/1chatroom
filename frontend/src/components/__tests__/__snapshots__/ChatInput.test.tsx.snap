// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatInput it renders 1`] = `
.c0 {
  background-color: rgba(250,157,38,0.0);
  padding: 12px;
}

.c0 >form {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 6px;
  outline: none;
}

<div
  className="c0"
>
  <form
    onSubmit={[Function]}
  >
    <input
      autoFocus={true}
      className="c1"
      data-testid="input"
      onChange={[Function]}
      value=""
    />
  </form>
</div>
`;
